rules/TypeDeclaration/Rector/ClassMethod/NumericReturnTypeFromStrictScalarReturnsRector.php

```diff
@@ -134,14 +157,27 @@ private function matchRootReturnWithExpr(ClassMethod|Function_ $functionLike): ?
         return null;
     }
 
+    private function isBinaryOpContainingNonTypedParam(
+        BinaryOp $binaryOp
+    ):bool {
+        if ($this->exprAnalyzer->isNonTypedFromParam($binaryOp->left)) {
+            return true;
+        }
+        if ($this->exprAnalyzer->isNonTypedFromParam($binaryOp->right)) {
+            return true;
+        }
+
+        return false;
+    }
+
     private function refactorBinaryOp(
         BinaryOp $binaryOp,
         ClassMethod|Function_ $functionLike
     ): null|Function_|ClassMethod {
         $leftType = $this->getType($binaryOp->left);
         $rightType = $this->getType($binaryOp->right);
 
-        if ($leftType instanceof IntegerType && $rightType instanceof IntegerType) {
+        if ($leftType->isInteger()->yes() && $rightType->isInteger()->yes()) {
```

Comment:
How can we verify this takes into account only direct integers? I'd prefer to be explicit and keep the original, instead of hoping it avoid docblock integer types, that are interprested as strict types by PHPStan.

Reply:
I thought I did by adding more tests. Do you have something concrete in mind which should be tested in addition?

I think we can't just add `instanceof IntegerRangeType` because there is also a union of integers in the game.

Reply:
I recall doc typed properties and external method/property calls are false positives. Could you add those too?

Reply:
I reverted my implementation changes for now, as you are right in that phpdoc typed properties are a problem.
these are a problem even before the changes I suggested before this initial PR.

I re-calibrated the goal of this PR to just add more test-cases for now, so this knowledge about edge cases of this rector is not lost.

----

I got the feeling the only real way this rector could work reliably is if it would use [treatPhpDocTypesAsCertain: false](https://phpstan.org/config-reference#treatphpdoctypesascertain). is this something we could configure on a per rector basis?

Reply:
I guess this setting would then affect all rector rules not just a one.

I wonder whether rector should have a "mode" in which it does not at all trust phpdoc, so it can be very sure about the things it is doing.